tests/dfencoder/test_ae_feature_selector.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+# SPDX-FileCopyrightText: Copyright (c) 2022-2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import pandas as pd
+import pytest
+from sklearn.datasets import load_iris
+
+from morpheus.models.dfencoder.ae_feature_selector import AutoencoderFeatureSelector
+
+
+@pytest.fixture
+def sample_data():
+    """Fixture to provide the Iris dataset for testing."""
+    iris = load_iris()
+    data = pd.DataFrame(data=iris.data, columns=iris.feature_names)
+    return data.to_dict(orient='records')
+
+
+def test_preprocess_data(sample_data):
+    """Test the preprocess_data method."""
+    selector = AutoencoderFeatureSelector(sample_data)
+    processed_data = selector.preprocess_data()
+    assert isinstance(processed_data, np.ndarray)
+    assert processed_data.shape[1] > 0  # Ensure columns exist after preprocessing
+
+
+def test_remove_low_variance(sample_data):
+    """Test removing low variance features."""
+    selector = AutoencoderFeatureSelector(sample_data)
+    processed_data = selector.preprocess_data()
+    reduced_data, mask = selector.remove_low_variance(processed_data)
+    assert reduced_data.shape[1] <= processed_data.shape[1]
+    assert isinstance(mask, np.ndarray)
```

Comment:
Is there anything else we can verify about the returned data besides the shape? Can we compare to a given known good result? For example, do we know which data points are "low variance" that we can verify they have been removed?